### PR TITLE
Add support for legacy scripting

### PR DESCRIPTION
**Pull request by eliangcs**
Adds `.legacyScriptingSource` to `AppSchema` and `.legacyProperties` to all operation schemas. 

`AppSchema.legacyScriptingSource` is a string property, holding the scripting code of a Web Builder app.

`*OperationSchema.legacyProperties` is an object that holds the properties, such as polling trigger URL, from a Web Builder app.

### CODE REVIEW

File: bin/export.js
Comment (eliangcs) [inline bin/export.js:8]: This is to remove the `Calling an asynchronous function without callback is deprecated` warning.

**Comment by eliangcs** [comment]
@bcooksey @BrunoBernardino @xavdid assigned to y'all to review as I'd like to get an approval from all of you before I merge this schema change.

File: lib/schemas/CodeSchema.js
Comment (BrunoBernardino) [inline lib/schemas/CodeSchema.js:8]: "any text" would break it though, unless you meant to add `// any text`, like a comment.

**Review by BrunoBernardino — APPROVED** [review]
Looks good! I have a question and also, aren't we adding the global code definition as well? Or you're planning on doing that in another PR, when it's clearer how that will interact with `core`?

File: lib/schemas/CodeSchema.js
Comment (eliangcs) [reply lib/schemas/CodeSchema.js:8]: I intend to let `'any test'` pass on the schema level. We don't (and don't want to) parse JS code just to validate, so any strings are allowed here. Does that make sense?

**Comment by eliangcs** [comment]
@BrunoBernardino yeah, I'm not sure how we're going to use global code definition in the UI yet, so I'll wait and do it in another PR.

File: lib/schemas/CodeSchema.js
Comment (BrunoBernardino) [reply lib/schemas/CodeSchema.js:8]: Makes total sense!

**Comment by xavdid** [comment]
Cool! So for the content itself i'm 👍. My big #recommendation is that you add a new doc annotation [here](https://github.com/zapier/zapier-platform-schema/blob/0020a8a4eebee08304d2e230fbc0b1eb02ca157e/lib/utils/buildDocs.js#L117) to the effect of "hide" or "remove" that keeps that property out of the docs entirely. Given that it's internal only, even with the bold "don't use this", it makes sense to me to keep it from the users entirely to save them from themselves.

**Comment by bcooksey** [comment]
@eliangcs What are the props we think will need to go into `legacyProperties`? I believe there is an analog for just about everything from Web Builder to the current schema. For example, the polling url should just becoming a perform like `perform: {url: '...'}`. Or in the case of a hook trigger, you have [performList](https://github.com/zapier/zapier-platform-schema/blob/0020a8a4eebee08304d2e230fbc0b1eb02ca157e/lib/schemas/BasicHookOperationSchema.js#L38-L42)

**Comment by eliangcs** [comment]
@xavdid thanks for the great idea! I'll push another commit making them hidden from the doc.

@bcooksey `legacyProperties` is there when the WB app has scripting and `perform` is a function source, such as `perform: {source: 'return z.legacyScripting.run(bundle, 'trigger', 'recipe');'}`. In this case,  `core` needs set the polling URL to `bundle` before calling `z.legacyScripting.run()`, but `perform` can't be used to hold the polling URL as it's a function source, so the polling URL must be stored somewhere else in the app definition for `core` to get and set it to `bundle`.

**Comment by bcooksey** [comment]
Got it. So any WB app that had scripting, we'll have to convert to code, meaning we lose those places to hold the URL and other data.

**Comment by eliangcs** [comment]
Closing this in favor of #60.

**Pull request closed without merging**
